### PR TITLE
fix(auth): TOTP contract fixes

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [ ] Added Unit Tests
- [ ] Added Integration Tests
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt [hunk 1]
```diff
@@ -12,18 +12,33 @@
...
+ * @param sharedSecret Secret code returned by the service to help setting up TOTP
+ * @param username username that will be used to construct the URI
+ */
 data class TOTPSetupDetails(
     val sharedSecret: String,
     val username: String
 ) {
+
+    /**
+     * Returns a TOTP setup URI that can help the customers avoid barcode scanning and use native password manager
```
Comment (mattcreaser) [inline core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt:31]: Should avoid saying "the customers" in API docs, as they are already the intended consumer of the documentation

File: core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt [hunk 2]
```diff
@@ -12,18 +12,33 @@
...
+     * to handle TOTP association.
+     *
+     * @param appName of TOTP manager
+     * @param accountName for TOTP manager
```
Comment (mattcreaser) [inline core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt:35]: Indicate the the default value here.

File: core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt [hunk 3]
```diff
@@ -12,18 +12,33 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
-package com.amplifyframework
+package com.amplifyframework.auth
 
 import android.net.Uri
 
+/**
+ * Details of TOTP Setup that help launch into TOTP manager.
```
Comment (mattcreaser) [inline core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt:20]: What is TOTP manager?

**Review by mattcreaser — COMMENTED** [review]
Few minor comments

File: core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt (hunk 1 above)
Comment (tylerjroach) [reply core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt:31]: Yeah I copied from swift. Will update

File: core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt (hunk 3 above)
Comment (tylerjroach) [reply core/src/main/java/com/amplifyframework/auth/TOTPSetupDetails.kt:20]: Authenticator apps on customers phone

**Pull request merged**